### PR TITLE
Creates intToBytes48 conversion

### PR DESCRIPTION
**Pull request by akhila-raju**
## PR description
Converts int to Bytes48 type.

### CODE REVIEW

File: util/src/main/java/tech/pegasys/artemis/util/bytes/Bytes48.java [hunk 1]
```diff
@@ -32,6 +34,23 @@
...
+   * @return      converted Bytes48
+   * @throws IllegalArgumentException if seed is a negative value.
+   */
+  static Bytes48 intToBytes48(int seed) {
+    checkArgument(seed > 0, "Expected positive seed but got %s", seed);
+    byte[] bytes = new byte[48];
+    for (int i = 0; i < 48; i++) {
+      bytes[i] = (byte) ((seed >> ((47 - i) * 8)) & 0xFF);
+    }
+    System.out.println(Arrays.toString(bytes));
```
Comment (jrhea) [inline util/src/main/java/tech/pegasys/artemis/util/bytes/Bytes48.java]: did you mean to keep a print statement in here?

Comment (akhila-raju) [reply util/src/main/java/tech/pegasys/artemis/util/bytes/Bytes48.java]: Whoops! Will remove

File: util/src/main/java/tech/pegasys/artemis/util/bytes/Bytes48.java [hunk 2]
```diff
@@ -32,6 +34,22 @@
...
+   *
+   * @param seed  converted
+   * @return      converted Bytes48
+   * @throws IllegalArgumentException if seed is a negative value.
+   */
+  static Bytes48 intToBytes48(int seed) {
+    checkArgument(seed > 0, "Expected positive seed but got %s", seed);
+    byte[] bytes = new byte[48];
+    for (int i = 0; i < 48; i++) {
+      bytes[i] = (byte) ((seed >> ((47 - i) * 8)) & 0xFF);
```
Comment (schroedingerscode) [inline util/src/main/java/tech/pegasys/artemis/util/bytes/Bytes48.java]: Per the Java spec (https://docs.oracle.com/javase/specs/jls/se8/html/jls-15.html#jls-15.19), the right-hand side of a shift operation is limited to 0-31 inclusive if the left-hand side is an int. This means for the first 44 iterations of the loop (0..43), the exact same action is performed:
     `(seed >> 31) & 0xFF`
which always yields one of 2 possible results: 
  -for integers >= 0: 0x0
  -for integers <0: 0xFF

Therefore, it would save 44 iterations to hardwire the uppermost bytes depending on the sign of the integer seed.

Comment (schroedingerscode) [inline util/src/main/java/tech/pegasys/artemis/util/bytes/Bytes48.java]: We might want to evaluate ByteOrder.nativeOrder() before we create this bytes object to ensure it matches the spec the system/program is expecting.

**Comment by schroedingerscode** [comment]
Other than performance concerns in previous comment, I see no issues with this implementation, as long as tests pass. The other option, instead of worrying about these edge cases ourselves, is to implement this type internally using a byte[] array and java.nio.ByteBuffer to do all the conversions. We just wrap and provide a (relatively) pretty interface for interacting with a Bytes## type without doing the conversions manually. 

@akhila-raju @jrhea - thoughts? ^^

File: util/src/main/java/tech/pegasys/artemis/util/bytes/Bytes48.java [hunk 3]
```diff
@@ -32,6 +34,22 @@
...
 
+  /**
+   * Converts int to Bytes48.
+   *
+   * @param seed  converted
+   * @return      converted Bytes48
+   * @throws IllegalArgumentException if seed is a negative value.
+   */
+  static Bytes48 intToBytes48(int seed) {
+    checkArgument(seed > 0, "Expected positive seed but got %s", seed);
```
Comment (schroedingerscode) [inline util/src/main/java/tech/pegasys/artemis/util/bytes/Bytes48.java]: Is there a reason we require a positive seed? 

If we don't expect and just don't want to handle/test negatives, that's fine, but the Java arithmetic right shift operator (>>) used in this implementation handles negatives just fine. The 2s complement value of negative numbers does end up in the byte array correctly.

**Comment by akhila-raju** [comment]
my only concern with that would be consistency with the other bytes types @schroedingerscode

**Comment by schroedingerscode** [comment]
Cool, works for me. Just hardwire the first 44 (0..43) bytes, update the javadoc for endianness clarity, and run spotlessApply and that works for me. We should make a task/issue so this gets sufficient test case coverage.

File: util/src/main/java/tech/pegasys/artemis/util/bytes/Bytes48.java (hunk 3 above)
Comment (akhila-raju) [reply util/src/main/java/tech/pegasys/artemis/util/bytes/Bytes48.java]: No, it's not necessary. I will change this to include negatives.

File: util/src/main/java/tech/pegasys/artemis/util/bytes/Bytes48.java [hunk 4]
```diff
@@ -32,6 +33,24 @@
...
   Bytes48 ZERO = wrap(new byte[48]);
 
+  /**
+   * Converts int to Bytes48 in Big Endian byte order.
+   *
+   * @param seed  converted
+   * @return      converted Bytes48
+   * @throws IllegalArgumentException if seed is a negative value.
+   */
+  static Bytes48 intToBytes48(int seed) {
```
Comment (schroedingerscode) [inline util/src/main/java/tech/pegasys/artemis/util/bytes/Bytes48.java]: If negative numbers are allowed, then if the seed is negative, the first 44 bytes must be hardwired to 0xFF. My suggestion would be something like this:

```
static Bytes48 intToBytes48(int seed) {
  byte[] bytes = new byte[48];
  if(seed < 0) {
    Arrays.fill(bytes, 0xFF);
  } else {
    Arrays.fill(bytes, 0x0);
  }
  for(int i = 44; i < 48; ++i) {
    bytes[i] = (byte) ((seed >> ((47-i) * 8)) & 0xFF);
  }
  return Bytes48.wrap(bytes);
}
```

File: util/src/main/java/tech/pegasys/artemis/util/bytes/Bytes48.java [hunk 5]
```diff
@@ -20,6 +20,7 @@
 import tech.pegasys.artemis.util.uint.UInt256Bytes;
 
 
+
 /**
  * A {@link BytesValue} that is guaranteed to contain exactly 48 bytes.
```
Comment (schroedingerscode) [inline util/src/main/java/tech/pegasys/artemis/util/bytes/Bytes48.java]: Specify in the javadoc that the int is loaded into the byte array in big endian order.

File: util/src/main/java/tech/pegasys/artemis/util/bytes/Bytes48.java (hunk 4 above)
Comment (akhila-raju) [reply util/src/main/java/tech/pegasys/artemis/util/bytes/Bytes48.java]: Argh - sorry being sick is clouding my brain. Will fix

Comment (schroedingerscode) [reply util/src/main/java/tech/pegasys/artemis/util/bytes/Bytes48.java]: Haha, you're excused. What's sick is I think about stuff like this for fun. 😛

**Comment by schroedingerscode** [comment]
Looks good. I created Issue #19 to track the need for additional testing.
